### ISSUE TITLE
Error due to generic type in property when using Python 3.7

### ISSUE THREAD

**Issue by treiher** [Parquery/pyicontract-lint#44]
```Python
from __future__ import annotations

class A:
    @property
    def b(self) -> list[int]:
        return []
```

```Console
% pyicontract-lint test.py
Traceback (most recent call last):
  File "/tmp/venv/lib/python3.7/site-packages/astroid/scoped_nodes.py", line 2682, in getitem
    methods = lookup(self, "__getitem__")
  File "/tmp/venv/lib/python3.7/site-packages/astroid/interpreter/dunder_lookup.py", line 50, in lookup
    return _class_lookup(node, name)
  File "/tmp/venv/lib/python3.7/site-packages/astroid/interpreter/dunder_lookup.py", line 58, in _class_lookup
    raise AttributeInferenceError(attribute=name, target=node)
astroid.exceptions.AttributeInferenceError: '__getitem__' not found on <ClassDef.list l.0 at 0x7f6f47bd4b00>.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/tmp/venv/lib/python3.7/site-packages/astroid/inference.py", line 401, in infer_subscript
    assigned = value.getitem(index_value, context)
  File "/tmp/venv/lib/python3.7/site-packages/astroid/scoped_nodes.py", line 2695, in getitem
    raise AstroidTypeError(node=self, context=context) from exc
astroid.exceptions.AstroidTypeError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/tmp/venv/bin/pyicontract-lint", line 8, in <module>
    sys.exit(main())
  File "/tmp/venv/lib/python3.7/site-packages/icontract_lint/main.py", line 70, in main
    return _main(args=args, stream=sys.stdout)
  File "/tmp/venv/lib/python3.7/site-packages/icontract_lint/main.py", line 49, in _main
    errors = icontract_lint.check_paths(paths=args.paths)
  File "/tmp/venv/lib/python3.7/site-packages/icontract/_checkers.py", line 641, in wrapper
    result = func(*args, **kwargs)
  File "/tmp/venv/lib/python3.7/site-packages/icontract_lint/__init__.py", line 621, in check_paths
    errs.extend(check_file(path=pth))
  File "/tmp/venv/lib/python3.7/site-packages/icontract/_checkers.py", line 641, in wrapper
    result = func(*args, **kwargs)
  File "/tmp/venv/lib/python3.7/site-packages/icontract_lint/__init__.py", line 588, in check_file
    lint_visitor.visit(node=tree)
  File "/tmp/venv/lib/python3.7/site-packages/icontract_lint/__init__.py", line 146, in visit
    return func(node)
  File "/tmp/venv/lib/python3.7/site-packages/icontract_lint/__init__.py", line 151, in visit_generic
    self.visit(child)
  File "/tmp/venv/lib/python3.7/site-packages/icontract_lint/__init__.py", line 146, in visit
    return func(node)
  File "/tmp/venv/lib/python3.7/site-packages/icontract_lint/__init__.py", line 530, in visit_ClassDef
    self.visit(child)
  File "/tmp/venv/lib/python3.7/site-packages/icontract_lint/__init__.py", line 146, in visit
    return func(node)
  File "/tmp/venv/lib/python3.7/site-packages/icontract_lint/__init__.py", line 415, in visit_FunctionDef
    inferred_returns = next(node.returns.infer())
  File "/tmp/venv/lib/python3.7/site-packages/astroid/node_classes.py", line 354, in infer
    yield from self._infer(context, **kwargs)
  File "/tmp/venv/lib/python3.7/site-packages/astroid/decorators.py", line 136, in raise_if_nothing_inferred
    yield next(generator)
  File "/tmp/venv/lib/python3.7/site-packages/astroid/decorators.py", line 100, in wrapped
    res = next(generator)
  File "/tmp/venv/lib/python3.7/site-packages/astroid/inference.py", line 408, in infer_subscript
    raise InferenceError(node=self, context=context) from exc
astroid.exceptions.InferenceError: Inference failed for <Subscript l.5 at 0x7f6f474654e0>.
```

### Versions

Python 3.7.3
pyicontract-lint    2.1.3
astroid             2.5.6

The issue does not occur when using Python 3.9.6.

**Comment by mristin** [comment]
Hi @treiher ,
Thanks for the submission! This seems like an issue purely related to astroid. How serious is this bug for you? Would you like me to submit an issue on their repository?

**Comment by treiher** [comment]
Not fixing this issue would mean that type hinting generics cannot be used in code which is analyzed by `pyicontract-lint` when Python >=3.7,<3.9 is used. It would be nice to have, but it is not critical for me.

**Comment by mristin** [comment]
@treiher ok, I'll see if I can refactor out astroid. I'm on holidays till end of August, so please apologize the delay.

**Comment by mristin** [comment]
@treiher I had another look and it would be actually quite hard to refactor out astroid from the pyicontract-lint. Hence it would be best if this issue could be fixed upstream in the astroid. Do you want me to create an issue in the astroid repo or you'd like to do it yourself?

**Comment by treiher** [comment]
> Do you want me to create an issue in the astroid repo or you'd like to do it yourself?

I think it makes more sense that you create the issue.

**Comment by mristin** [comment]
@treiher I'll close this issue on pyicontract-lint as it is pretty certainly a problem with astroid. Please feel free to re-open if we can not get it eventually fixed on the upstream.

**Comment by mristin** [comment]
@treiher thanks to input from the team PyCQA I was able to hopefully patch this issue.

Would you mind testing that the pull request #45 fixes the problem on your side as well?

**Comment by mristin** [comment]
@treiher I've just released a new version, 2.1.4, that should fix your issue. I'll close the issue for now. Please let me know if the problem persists.

**Comment by treiher** [comment]
@mristin Version 2.1.4 solved the issue for me. Thanks!
